location-search/SearchClient.ts

```diff
@@ -0,0 +1,148 @@
+ *
+ * @param place A given {@link Place} from Geo.searchByText.
+ * @returns The parameter converted into a {@link TiFLocation}.
+ */
+export const awsPlaceToTifLocation = (place: Place) => {
+  if (!place.geometry) return undefined
+  return {
+    coordinates: {
+      latitude: place.geometry.point[1],
+      longitude: place.geometry.point[0]
+    },
+    placemark: {
+      name: place.label?.split(", ", 1)[0],
+      country: place.country,
+      postalCode: place.postalCode,
+      street: place.street,
+      streetNumber: place.addressNumber,
+      region: place.region,
+      isoCountryCode: "US",
+      city: place.municipality
+    }
+  }
+}
+
+/**
+ * A function that takes in a query, and uses a search function to obtain results, then converts the data
+ * into a usable set of {@link TiFLocation}s.
+ *
+ * @param query A {@link LocationsSearchQuery} given for the search function to use.
+ * @param center An optional {@link LocationCoordinate2D} given for the search to center in where it should be searching.
+ * @returns A Promise of an array of {@link TiFLocation}s.
+ */
+export const awsGeoSearchLocations = async (
+  query: LocationsSearchQuery,
+  center?: LocationCoordinate2D
+) => {
+  const options = {
+    maxResults: 10,
+    biasPosition: center ? [center?.longitude, center?.latitude] : undefined
+  }
+  const geoSearchResults = await Geo.searchByText(query.toString(), options)
+  return ArrayUtils.compactMap(geoSearchResults, awsPlaceToTifLocation)
+}
+
+/**
+ * A function that pulls the 10 most recent locations stored in AsyncStorage, and maps out their data
+ * into a set of {@link LocationSearchResult}s.
+ * @returns A Promise of an array of {@link LocationSearchResult}s.
+ */
+export const searchRecentLocations = (): Promise<LocationSearchResult[]> => {
+  return asyncStorageLoadRecentLocations(10).then((results) =>
+    results.map((area) => ({ ...area, isRecentLocation: true }))
+  )
+}
+
+/**
+ * A function that uses a provided query, with both its provided searchFunction to obtain {@link TiFLocation}s,
+ * and {@link asyncStorageLoadSpecificRecentLocations} on their coordinates to obtain async storage's information
+ * on whether those locations are recent.
+ * This involves combining the data from both processes.
+ *
+ * @param query A {@link LocationsSearchQuery} given for the search function to use.
+ * @param searchFunction A function that takes in a query + an optional center, then converts the data into a usable set of {@link TiFLocation}s.
+ * @param center An optional {@link LocationCoordinate2D} given for the search to center in where it should be searching.
+ * @returns A Promise of an array of {@link LocationSearchResult}s.
+ */
+
+export const searchWithRecentAnnotations = async (
+  query: LocationsSearchQuery,
+  searchFunction: (
+    query: LocationsSearchQuery,
+    center?: LocationCoordinate2D
+  ) => Promise<TiFLocation[]>,
+  center?: LocationCoordinate2D
```

Comment:
Nit: would move this above the `searchFunction` parameter.

Reply:
Can't, center is an optional parameter

Reply:
Instead, could make center 
```center: LocationCoordinate2D | undefined```